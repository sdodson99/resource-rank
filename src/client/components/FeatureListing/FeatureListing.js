import React from 'react';
import PropTypes from 'prop-types';
import styles from './FeatureListing.module.css';
import Feature from '../Feature/Feature';

const FeatureListing = ({ features }) => {
  const featureItems = features.map((f) => (
    <div className={styles.FeatureItem} key={f.title}>
      <div className="content-container flex-grow">
        <Feature description={f.description} imageSrc={f.imageSrc} />
      </div>
    </div>
  ));

  return (
    <div className={styles.FeatureListing} data-testid="FeatureListing">
      {featureItems}
    </div>
  );
};

FeatureListing.propTypes = {
  features: PropTypes.arrayOf(
    PropTypes.shape({
      description: PropTypes.node.isRequired,
      imageSrc: PropTypes.string.isRequired,
    })
  ),
};

FeatureListing.defaultProps = {
  features: [],
};

export default FeatureListing;
