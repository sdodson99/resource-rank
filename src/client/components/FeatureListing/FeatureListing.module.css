.FeatureListing {
}

.FeatureItem {
  min-height: 300px;

  @apply py-10 flex items-center;
}

.FeatureItem:nth-child(even) {
  @apply bg-gray-50;
}
