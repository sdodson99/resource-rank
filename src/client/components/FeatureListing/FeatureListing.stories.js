import React from 'react';
import FeatureListing from './FeatureListing';

export default {
  component: FeatureListing,
  title: 'Components/FeatureListing',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <FeatureListing {...args} />;

export const Default = Template.bind({});
Default.args = {
  features: [
    {
      title: 'Instant',
      description: 'Super fast!',
      imageSrc: '/img/instantly-find.svg',
    },
    {
      title: 'Explore',
      description: 'Explore!',
      imageSrc: '/img/explore.svg',
    },
    {
      title: 'Rate',
      description: 'Rate!',
      imageSrc: '/img/rate-resources.svg',
    },
  ],
};

export const Single = Template.bind({});
Single.args = {
  features: [
    {
      title: 'Instant',
      description: 'Super fast!',
      imageSrc: '/img/instantly-find.svg',
    },
  ],
};

export const Empty = Template.bind({});
Empty.args = {};
