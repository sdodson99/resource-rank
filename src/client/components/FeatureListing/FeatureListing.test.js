import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import FeatureListing from './FeatureListing';

describe('<FeatureListing />', () => {
  test('it should mount', () => {
    render(<FeatureListing />);

    const featureListing = screen.getByTestId('FeatureListing');

    expect(featureListing).toBeInTheDocument();
  });

  describe('with features', () => {
    let features;

    beforeEach(() => {
      features = [
        {
          title: '1',
          description: '123',
          imageSrc: 'src',
        },
        {
          title: '2',
          description: '456',
          imageSrc: 'src',
        },
      ];
    });

    it('should render correctly', () => {
      const tree = renderer
        .create(<FeatureListing features={features} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('without features', () => {
    it('should render correctly', () => {
      const tree = renderer.create(<FeatureListing />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
