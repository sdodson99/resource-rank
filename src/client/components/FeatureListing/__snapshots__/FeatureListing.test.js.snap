// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeatureListing /> with features should render correctly 1`] = `
<div
  className="FeatureListing"
  data-testid="FeatureListing"
>
  <div
    className="FeatureItem"
  >
    <div
      className="content-container flex-grow"
    >
      <div
        className="Feature"
        data-testid="Feature"
      >
        <div
          className="FeatureDescription"
        >
          123
        </div>
        <img
          alt="Feature Image"
          className="FeatureImage"
        />
      </div>
    </div>
  </div>
  <div
    className="FeatureItem"
  >
    <div
      className="content-container flex-grow"
    >
      <div
        className="Feature"
        data-testid="Feature"
      >
        <div
          className="FeatureDescription"
        >
          456
        </div>
        <img
          alt="Feature Image"
          className="FeatureImage"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<FeatureListing /> without features should render correctly 1`] = `
<div
  className="FeatureListing"
  data-testid="FeatureListing"
/>
`;
